### PR TITLE
💄 Add setting page

### LINKED ISSUE #5: As a User I want to add my start- and end-location and date
- [x] Add Start-Location
- [x] Add End-Location
- [x] Add first day
- [x] Add last day

### LINKED ISSUE #8: As a user, I want to close the site and see and edit my plan
- [ ] Add LocalStorage

### PR DESCRIPTION
**Pull request by m-wiemers**
Fix #8 
Fix #5 

- Add Setting Page
- Add DayCounter
- Add LocalStorage

### CODE REVIEW

File: components/DateInput/DateInput.module.css [hunk 1]
```diff
@@ -3,6 +3,10 @@
 }
 
 .date {
+  justify-self: center;
+  margin-top: 0.5em;
+  min-width: 350px;
+  max-width: 90vw;
```
Comment (lmachens) [inline components/DateInput/DateInput.module.css]: Imagine you like to use `DateInput` in a form with a different layout.
Viewport relative widths are very problematic for components, instead, it should be controlled by the parent container (e.g. form).
Same issue in `SelectCity`

File: pages/settings.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        <SelectCity
+          label={"start location"}
+          value={bigCities}
+          onSelect={handleStartLocationChange}
+        />
+
+        <SelectCity
+          label={"end location"}
+          value={bigCities}
+          onSelect={() => <p></p>}
```
Comment (lmachens) [inline pages/settings.tsx]: What is `<p></p>` for?

File: pages/settings.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          value={"2021-04-01"}
+          onDateSelect={() => <p></p>}
+        />
+
+        <div>
+          <Button label={"continue"} onButtonClick={() => <p></p>} />
+        </div>
+      </main>
+      <footer className={styles.footer}>
+        <Footer isActive={"settings"} />
```
Comment (lmachens) [inline pages/settings.tsx]: If I focus on this line, I would ask myself what is `isActive`?
When is a `Footer` active? When is a `Footer` inactive?

**Review by lmachens — COMMENTED** [review]
A description would be helpful to understand what you tried to achieve in this pull request and what you like to do later. The settings page is not complete (which is not an issue), but I would expect a full working page when I read your request.

Note that GitHub CI added some comments to your request too (unused variables).

File: components/dayCounter/DayCounter.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  label,
+  lastDate,
+  startDate,
+  secondLabel,
+}: DayCounterProps) {
+  const [dayCount, setDayCount] = useState(null);
+
+  useEffect(() => {
+    const firstDate = new Date(startDate);
+    const lastDatee = new Date(lastDate);
```
Comment (RonjaMeinecke) [inline components/dayCounter/DayCounter.tsx]:
```suggestion
    const lastDate = new Date(lastDate);
```

Comment (RonjaMeinecke) [inline components/dayCounter/DayCounter.tsx]:
```suggestion
    const endDate = new Date(endDate);
```

File: components/footer/Footer.tsx [hunk 5]
```diff
@@ -5,10 +5,10 @@ import CalenderIcon from "../icons/CalenderIcon";
 import SettingIcon from "../icons/SettingIcon";
 
 export type FooterProps = {
-  isActive: "list" | "calender" | "map" | "settings";
+  activeButton: "list" | "calender" | "map" | "settings";
```
Comment (RonjaMeinecke) [inline components/footer/Footer.tsx]:
```suggestion
  activeButton: "list" | "calendar" | "map" | "settings";
```

File: styles/Settings.module.css [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+.main {
+  text-align: center;
+  margin-top: 1em;
+  display: grid;
+  justify-items: center;
+  row-gap: 1em;
+}
+
+.main :only-child {
+  min-width: 320px;
```
Comment (RonjaMeinecke) [inline styles/Settings.module.css:16]: do you really want to use px here?

File: styles/Settings.module.css [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  max-width: 90vw;
+}
+
+.button {
+  margin-top: 2em;
+  justify-self: center;
+  max-width: 350px;
```
Comment (RonjaMeinecke) [inline styles/Settings.module.css:23]: do you really want to use px here?

**Review by RonjaMeinecke — APPROVED** [review]
great progress mate ✨

**Review by mrana17 — APPROVED** [review]
you are smashing it! 💪

**Review by SilviaBarcellini — APPROVED** [review]
Brilliant! 👍

**Pull request merged**